File: LocalisationMateriel/src/java/metier/MetrologueServiceImpl.java
package metier;

import java.util.ArrayList;
import java.util.List;
import physique.MetrologueDataService;
import physique.PhysiqueFactory;

public class MetrologueServiceImpl implements MetrologueService {

    private final MetrologueDataService metrologueDataSrv;

    MetrologueServiceImpl() {
        metrologueDataSrv = PhysiqueFactory.getMetrologueDataService();
    }

    @Override
    public Metrologue add(Metrologue metrologue) {
        Metrologue met = null;
        try {
            if (!metrologue.getLogin().isEmpty() && !metrologue.getPassword().isEmpty()) {
                met = metrologueDataSrv.add(metrologue);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return met;
    }

    @Override
    public boolean remove(Metrologue metrologue) {
        boolean execute = false;
        try {
            if (metrologue.getId() > 0) {
                execute = metrologueDataSrv.remove(metrologue);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return execute;
    }

    @Override
    public boolean update(Metrologue metrologue) {
        boolean execute = false;
        try {
            if (metrologue.getId() > 0) {
                if (!metrologue.getLogin().isEmpty() && !metrologue.getPassword().isEmpty()) {
                    execute = metrologueDataSrv.update(metrologue);
                }
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return execute;
    }

    @Override
    public List<Metrologue> getAll(int debut, int parPage) {
        List<Metrologue> met = new ArrayList<Metrologue>();
        try {
            if (debut >= 0 && parPage > 0) {
                met = metrologueDataSrv.getAll(debut, parPage);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return met;
    }

    @Override
    public int getCount() {
        int count = 0;
        try {
            count = metrologueDataSrv.getCount();
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return count;
    }

    @Override
    public Metrologue getById(long id) {
        Metrologue met = null;
        try {
            if (id > 0) {
                met = metrologueDataSrv.getById(id);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return met;
    }

    @Override
    public Metrologue getByLogin(String login) {
        Metrologue met = null;
        try {
            if (!login.isEmpty()) {
                met = metrologueDataSrv.getByLogin(login);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return met;
    }

    @Override
    public List<String> getListLogin(int debut, int parPage) {
        List<String> listStr = new ArrayList<String>();
        try {
            if (debut >= 0 && parPage > 0) {
                listStr = metrologueDataSrv.getListLogin(debut, parPage);
            }
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return listStr;
    }

    @Override
    public int getCountListLogin() {
        int count = 0;
        try {
            count = metrologueDataSrv.getCountListLogin();
        } catch (Exception ex) {
            System.out.println(ex.getMessage());
        }
        return count;
    }

}

File: AppMateriel/metier/metrologueserviceimpl.cpp
#include "metrologueserviceimpl.h"

Metier::MetrologueServiceImpl::MetrologueServiceImpl()
{
    metrologueDataSrv = PhysiqueFactory::getMetrologueDataServiceImpl();
}

Metrologue Metier::MetrologueServiceImpl::add(Metrologue metrologue)
{
    Metrologue met;
    if(!metrologue.getLogin().isEmpty() && !metrologue.getPassword().isEmpty()){
        met = metrologueDataSrv->add(metrologue);
    }
    return met;
}

bool Metier::MetrologueServiceImpl::remove(Metrologue metrologue)
{
    bool execute = false;
    if(metrologue.getId() > 0){
        execute = metrologueDataSrv->remove(metrologue);
    }
    return execute;
}

bool Metier::MetrologueServiceImpl::update(Metrologue metrologue)
{
    bool execute = false;
    if(metrologue.getId() > 0){
        if(!metrologue.getLogin().isEmpty() && !metrologue.getPassword().isEmpty()){
            execute = metrologueDataSrv->update(metrologue);
        }
    }
    return execute;
}

QList<Metrologue> Metier::MetrologueServiceImpl::getAll(int debut, int parPage)
{
    QList<Metrologue> met;
    if(debut >= 0 && parPage > 0){
        met = metrologueDataSrv->getAll(debut, parPage);
    }
    return met;
}

int Metier::MetrologueServiceImpl::getCount()
{
    return metrologueDataSrv->getCount();
}

Metrologue Metier::MetrologueServiceImpl::getById(long id)
{
    Metrologue met;
    if(id > 0){
        met = metrologueDataSrv->getById(id);
    }
    return met;
}

Metrologue Metier::MetrologueServiceImpl::getByLogin(QString login)
{
    Metrologue met;
    if(!login.isEmpty()){
        met = metrologueDataSrv->getByLogin(login);
    }
    return met;
}

QList<QString> Metier::MetrologueServiceImpl::getListLogin(int debut, int parPage)
{
    QList<QString> listStr;
    if(debut >= 0 && parPage > 0){
        listStr = metrologueDataSrv->getListLogin(debut, parPage);
    }
    return listStr;
}

int Metier::MetrologueServiceImpl::getCountListLogin()
{
    int count = metrologueDataSrv->getCountListLogin();
    return count;
}
